pkg/mcp/mcp.go

```diff
@@ -2,362 +2,164 @@ package mcp
+	// eg. "config volumes add" -> "func://help/config/volumes/add")
+	i.AddResource(newHelpResource(s, "Help", "help for the command root", ""))
+	i.AddResource(newHelpResource(s, "Create Help", "help for 'create'", "create"))
+	i.AddResource(newHelpResource(s, "Build Help", "help for 'build'", "build"))
+	i.AddResource(newHelpResource(s, "Deploy Help", "help for 'deploy'", "deploy"))
+	i.AddResource(newHelpResource(s, "List Help", "help for 'list'", "list"))
+
+	i.AddResource(newHelpResource(s, "Volumes Help", "general help for volumes", "config", "volumes"))
+	i.AddResource(newHelpResource(s, "Volumes Add Help", "help for 'config volumes add'", "config", "volumes", "add"))
+	i.AddResource(newHelpResource(s, "Volumes Remove Help", "help for 'config volumes remove'", "list", "volumes", "remove"))
+
+	i.AddResource(newHelpResource(s, "Labels Help", "general help for labels", "config", "labels"))
+	i.AddResource(newHelpResource(s, "Labels Add Help", "help for 'config labels add'", "config", "labels", "add"))
+	i.AddResource(newHelpResource(s, "Labels Remove Help", "help for 'config labels remove'", "list", "labels", "remove"))
+
+	i.AddResource(newHelpResource(s, "Envs Help", "general help for environment variables", "config", "envs"))
+	i.AddResource(newHelpResource(s, "Envs Add Help", "help for 'config envs add'", "config", "envs", "add"))
+	i.AddResource(newHelpResource(s, "Envs Remove Help", "help for 'config envs remove'", "list", "envs", "remove"))
+
+	s.impl = i
+
+	return s
+}
 
-	mcpServer.AddPrompt(mcp.NewPrompt("help",
-		mcp.WithPromptDescription("help prompt for the root command"),
-	), handleRootHelpPrompt)
+// Start the MCP server using the configured transport
+func (s *Server) Start(ctx context.Context, writeEnabled bool) error {
+	s.readonly = !writeEnabled
+	return s.impl.Run(ctx, s.transport)
+}
 
-	mcpServer.AddPrompt(mcp.NewPrompt("cmd_help",
-		mcp.WithPromptDescription("help prompt for a specific command"),
-		mcp.WithArgument("cmd",
-			mcp.ArgumentDescription("The command for which help is requested"),
-			mcp.RequiredArgument(),
-		),
-	), handleCmdHelpPrompt)
+// For now the executor is a simple run of the command "func" or "kn func"
+// etc.  This should be replaced with a direct integration with the functions
+// client API.
+type defaultExecutor struct {
+	s *Server
+}
 
-	mcpServer.AddPrompt(mcp.NewPrompt("list_templates",
-		mcp.WithPromptDescription("prompt to list available function templates"),
-	), handleListTemplatesPrompt)
+func (e defaultExecutor) Execute(ctx context.Context, subcommand string, args ...string) ([]byte, error) {
+	// Parse prefix: "func" or "kn func" -> ["func"] or ["kn", "func"]
+	cmdParts := strings.Fields(e.s.prefix)
+	cmdParts = append(cmdParts, subcommand)
+	cmdParts = append(cmdParts, args...)
 
-	return &MCPServer{
-		server: mcpServer,
-	}
+	cmd := exec.CommandContext(ctx, cmdParts[0], cmdParts[1:]...)
+	// Commands always execute in current working directory
```

Comment:
It's the default behavior since `cmd.Dir` is not set. I would either remove the comment or make it more precise:

```suggestion
	// cmd.Dir not set - inherits process working directory which is the current working directory
```